java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/JsonDeserializer.java

```diff
@@ -49,63 +49,82 @@ public Iterator<SinkRow> deserialize(Object payload) {
                 .iterator();
     }
 
-    private static Object validateJsonDataTypes(Data.DataType.TypeName typeName, Object value) {
-        if (value instanceof Double
-                && (Double) value % 1 == 0
-                && typeName != Data.DataType.TypeName.DOUBLE
-                && typeName != Data.DataType.TypeName.FLOAT) {
-            return (int) (double) value;
+    private static Long castLong(Object value) {
+        if (value instanceof Integer) {
+            return ((Integer) value).longValue();
+        } else if (value instanceof Double) {
```

Comment:
I see the places calling `castLong` are INT16, INT32, and INT64. Why should we cover the case of Double and Float?

Reply:
The json library we are using will convert a json value such `1` into a double. To be compatible with it, we may also check the case of double and float. We have added a check on whether the floating number represents an integer. For example, if we get a `1.1` in `castLong`, we will get an error.